SignUpValidation/index.js

```diff
@@ -0,0 +1,70 @@
+module.exports = async function (context, req) {
+  context.log("JavaScript HTTP trigger function processed a request v1.");
+  const API_VERSION = "1.0.0";
+
+  // Parse Basic Auth username and password
+  var header = req.headers["authorization"] || "", // get the header
+    token = header.split(/\s+/).pop() || "", // and the encoded auth token
+    auth = new Buffer.from(token, "base64").toString(), // convert from base64
+    parts = auth.split(/:/), // split on colon
+    username = parts[0],
+    password = parts[1];
+
+  // Check for HTTP Basic Authentication, return HTTP 401 error if invalid credentials.
+  if (
+    username !== process.env["BASIC_AUTH_USERNAME"] ||
+    password !== process.env["BASIC_AUTH_PASSWORD"]
+  ) {
+    context.res = {
+      status: 401,
+    };
+    context.log("Invalid Authentication");
+    return;
+  }
+
+  // If input data is null, return error.
+  const INVALID_REQUEST = {
+    status: 400,
+    body: {
+      version: API_VERSION,
+      code: "INVALID_REQUEST",
+    },
+  };
+
+  if (!(req.body && req.body.email)) {
+    context.res = INVALID_REQUEST;
+    context.log("Invalid Request");
+    return;
+  }
+
+  // Log the request body
+  context.log(`Request body: ${JSON.stringify(req.body)}`);
+
+  // Get domain of email address
+  const domain = req.body.email.split("@")[1];
+  const allowedDomains = ["ioet.com"];
+
+  context.log("Validation: ", allowedDomains.includes(domain.toLowerCase()));
+  // Check that the domain of the email is from a specific other tenant
+  if (!allowedDomains.includes(domain.toLowerCase())) {
```

Comment:
You're only looking for `ioet.com` as a valid domain?
I don't think so, you should find a way to make sure the token is correct and it is not expired.

Reply:
At this point we didn't have the token yet.
Since is before the user is created, and it's validating the email with which the user is trying to log in with the google provider.